src/Networks/Network.ts

```diff
@@ -0,0 +1,323 @@
+        }
+        if (!Validate.isUndefined(props.dhcpd_relay_enabled)) {
+            this.dhcpd_relay_enabled = props.dhcpd_relay_enabled;
+        }
+        if (!Validate.isUndefined(props.dhcpd_boot_enabled)) {
+            this.dhcpd_boot_enabled = props.dhcpd_boot_enabled;
+        }
+        if (!Validate.isUndefined(props.igmp_proxy_downstream)) {
+            this.igmp_proxy_downstream = props.igmp_proxy_downstream;
+        }
+        if (!Validate.isUndefined(props.upnp_lan_enabled)) {
+            this.upnp_lan_enabled = props.upnp_lan_enabled;
+        }
+        if (!Validate.isUndefined(props.dhcpd_ntp_enabled)) {
+            this.dhcpd_ntp_enabled = props.dhcpd_ntp_enabled;
+        }
+        if (!Validate.isUndefined(props.mdns_enabled)) {
+            this.mdns_enabled = props.mdns_enabled;
+        }
+        if (!Validate.isUndefined(props.lte_lan_enabled)) {
+            this.lte_lan_enabled = props.lte_lan_enabled;
+        }
+        if (!Validate.isUndefined(props.dhpcd_tftp_server)) {
+            this.dhpcd_tftp_server = props.dhpcd_tftp_server;
+        }
+        if (!Validate.isUndefined(props.auto_scale_enabled)) {
+            this.auto_scale_enabled = props.auto_scale_enabled;
+        }
+        if (!Validate.isUndefined(props.vlan)) {
+            this.vlan = props.vlan;
+        }
+        if (!Validate.isUndefined(props.l2tp_allow_weak_ciphers)) {
+            this.l2tp_allow_weak_ciphers = props.l2tp_allow_weak_ciphers;
+        }
+        if (!Validate.isUndefined(props.l2tp_local_wan_ip)) {
+            this.l2tp_local_wan_ip = props.l2tp_local_wan_ip;
+        }
+        if (!Validate.isUndefined(props.vpn_type)) {
+            this.vpn_type = props.vpn_type;
+        }
+        if (!Validate.isUndefined(props.radiusprofile_id)) {
+            this.radiusprofile_id = props.radiusprofile_id;
+        }
+        if (!Validate.isUndefined(props.dhpcd_wins_enabled)) {
+            this.dhpcd_wins_enabled = props.dhpcd_wins_enabled;
+        }
+        if (!Validate.isUndefined(props.require_mschapv2)) {
+            this.require_mschapv2 = props.require_mschapv2;
+        }
+        if (!Validate.isUndefined(props.l2tp_interface)) {
+            this.l2tp_interface = props.l2tp_interface;
+        }
+        if (!Validate.isUndefined(props.exposed_to_site_vpn)) {
+            this.exposed_to_site_vpn = props.exposed_to_site_vpn;
+        }
+        if (!Validate.isUndefined(props.x_ipsec_pre_shared_key)) {
+            this.x_ipsec_pre_shared_key = props.x_ipsec_pre_shared_key;
+        }
+
+        return this;
+    }
+
+    public async update(props: Partial<INetwork>): Promise<void> {
+        if (!Validate.isDefinedNotNull(props._id)) {
+            throw new Error('Cannot update network without _id');
+        }
+
+        return (
+            await this.instance.put('/rest/networkconf/:id', props, {
+                urlParams: {
+                    site: this.site.name,
```

Comment:
```suggestion
```
site not needed ( added automatically by parent site )

Reply:
Got it